src/Lecture3.hs

```diff
@@ -122,10 +129,12 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  (<>) :: Reward -> Reward -> Reward
+  Reward ga sa <> Reward gb sb = Reward (ga <> gb) (sa || sb)
 
 instance Monoid Reward where
-
+  mempty :: Reward
+  mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
  mempty = Reward mempty False
```